src/main/java/seedu/weme/ui/StatsPanel.java

```diff
@@ -14,44 +16,74 @@
 
 import seedu.weme.model.ReadOnlyWeme;
 import seedu.weme.model.meme.Meme;
-import seedu.weme.statistics.TagWithCount;
+import seedu.weme.statistics.TagWithStats;
 
 /**
  * Panel containing the statistical data about memes in Weme.
  */
 public class StatsPanel extends UiPart<Region> {
     private static final String FXML = "StatsPanel.fxml";
-    private static final String PIE_CHART_TITLE = "Tag usage";
+    private static final String COUNT_CHART_TITLE = "Tag usage";
+    private static final String LIKE_CHART_TITLE = "Likes per Tag";
     private static final int LABEL_LINE_LENGTH = 10;
 
     @FXML
-    private PieChart piechart;
+    private PieChart tagCountChart;
+    @FXML
+    private PieChart tagLikeChart;
 
     public StatsPanel(ReadOnlyWeme weme) {
         super(FXML);
-        generatePieChart(weme);
-        weme.getMemeList().addListener((ListChangeListener<Meme>) change -> generatePieChart(weme));
+        renderCharts(weme);
+    }
+
+    /**
+     * Renders the charts with Chart generator methods.
+     */
+    private void renderCharts(ReadOnlyWeme weme) {
+        generateTagCountChart(weme);
+        generateTagLikeChart(weme);
+        weme.getMemeList().addListener((ListChangeListener<Meme>) change -> renderCharts(weme));
+        weme.getStats().getObservableLikeData().addListener((MapChangeListener<String, Integer>) change ->
```

Comment:
doesn't this mean that you are adding a new listener every time the meme list / likedata is modified? would be better to just add a listener at the start. renderChart shouldnt be adding listeners again.

Reply:
Oh no.. I moved it to abstract it away
Will move it out.